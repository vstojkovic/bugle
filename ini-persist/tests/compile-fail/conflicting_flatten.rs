use ini_persist::{LoadProperty, Result};

#[derive(LoadProperty)]
struct Foo {
    argle: u8,
}

#[derive(LoadProperty)]
struct Bar {
    #[ini(flatten)]
    #[ini(rename = "Bargle")]
    bargle: Foo,

    #[ini(flatten)]
    #[ini(key_format = "{name}{prefix}")]
    glop: Foo,

    #[ini(flatten)]
    #[ini(parse_with = my_parse)]
    glyf: Foo,

    #[ini(rename = "OlleBolle")]
    #[ini(flatten)]
    olle_bolle: Foo,

    #[ini(key_format = "{name}{prefix}")]
    #[ini(flatten)]
    snop: Foo,

    #[ini(parse_with = my_parse)]
    #[ini(flatten)]
    snyf: Foo,
}

fn my_parse(value: &str) -> Result<Foo> {
    todo!()
}

fn main() {
}
