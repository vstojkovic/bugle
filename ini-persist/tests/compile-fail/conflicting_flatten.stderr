error: cannot rename a flattened field
  --> tests/compile-fail/conflicting_flatten.rs:12:11
   |
12 |     #[ini(rename = "Bargle")]
   |           ^^^^^^^^^^^^^^^^^

error: cannot define a key format for a flattened field
  --> tests/compile-fail/conflicting_flatten.rs:16:11
   |
16 |     #[ini(key_format = "{name}{prefix}")]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot define a load function for a flattened field
  --> tests/compile-fail/conflicting_flatten.rs:20:11
   |
20 |     #[ini(parse_with = my_parse)]
   |           ^^^^^^^^^^^^^^^^^^^^^

error: cannot flatten a renamed field
  --> tests/compile-fail/conflicting_flatten.rs:24:11
   |
24 |     #[ini(flatten)]
   |           ^^^^^^^

error: cannot flatten a field with a defined key format
  --> tests/compile-fail/conflicting_flatten.rs:28:11
   |
28 |     #[ini(flatten)]
   |           ^^^^^^^

error: cannot flatten a field with a defined load function
  --> tests/compile-fail/conflicting_flatten.rs:32:11
   |
32 |     #[ini(flatten)]
   |           ^^^^^^^
